Formula/l/lexido.rb

```diff
@@ -0,0 +1,21 @@
+class Lexido < Formula
+  desc "Innovative assistant for the command-line"
+  homepage "https://github.com/micr0-dev/lexido"
+  url "https://github.com/micr0-dev/lexido.git", tag: "v1.3.1", revision: "36f02d699f2709f0bd51a5f119e0670845b4ad9f"
+  license "AGPL-3.0-or-later"
+
+  depends_on "go" => :build
+
+  def install
+    # Build the binary using the default go build command
+    system "go", "build", *std_go_args
+  end
+
+  test do
+    # Run the `lexido` command with the "test" and the local argument
+    output = shell_output("#{bin}/lexido -l \"test\" 2>&1", 1).strip
+
+    # Check if the error message indicates that ollama is not installed
+    assert_match "Error initializing ollama: ollama not installed", output
```

Comment:
Can it be tested with the [`ollama`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/o/ollama.rb) formula then? You can add that as a test dependency.

```ruby
depends_on "ollama" => :test
```

Reply:
The test I need to do with ollama is to see if it fails because if I do the lexido -l “test” it will give me a different response every time. that's if Ollama is fully installed and set I can try a new test with the Ollama if you think that will be better

Reply:
I don't completely understand how homebrew formulas work, or what you are saying, but I just want to say that **ollama is NOT a dependency** of lexido, it is an optional addition for people who would like to use local LLMs on their machines.

Reply:
Yes, thanks for the clarification. I was only suggesting that `lexido` can be tested with `ollama`. Test dependencies do not get installed with plain `brew install`.

If you have suggestions on alternative ways to test the installation without `ollama`, we can try that too.

Reply:
Well an API key is needed for the main functionality of the program but what we can test is a failed attempt to connect with an invalid API key, for the simplicity of the test it can be set by setting a temporary ENV variable with the wrong key. And then we expect an invalid key error to be returned

Reply:
> Well an API key is needed for the main functionality of the program but what we can test is a failed attempt to connect with an invalid API key, for the simplicity of the test it can be set by setting a temporary ENV variable with the wrong key. And then we expect an invalid key error to be returned

I will change the test to a fake API key as you sent the right name of how the key is stored in the issue on your repo and test to see what I get.

Reply:
ok I cant get the fake API key to work its has the same error no matter what i do ive made 5 versions of the test block nothing changes the error 

`  test do
    # Set a fake API key for testing purposes
    ENV["GOOGLE_AI_KEY"] = "12309863243987"

    # Run the `lexido` command
    output = shell_output("#{bin}/lexido", 1).strip

    assert_equal 1, $?.exitstatus
  end` -> 1st 
  
  `  test do
    # Set a fake API key for testing purposes
    ENV["GOOGLE_AI_KEY"] = "12309863243987"

    # Run the `lexido` command
    output = shell_output("#{bin}/lexido", 1).strip

    # Check if the error message or relevant parts are present in the output
    assert_match "Connecting...", output
    assert_match "API key not valid. Please pass a valid API key", output
    assert_match "INVALID_ARGUMENT", output
  end` -> 2nd
  
  `  test do
    # Set a fake API key for testing purposes
    ENV["GOOGLE_AI_KEY"] = "12309863243987"

    # Run the `lexido` command with the "test" argument
    output = shell_output("#{bin}/lexido \"test\"", 1).strip

    # Check if the error message or relevant parts are present in the output
    assert_match "error details: googleapi: got HTTP response code 400 with bod", output
    assert_match "API key not valid. Please pass a valid API key", output
  end`-> 3rd
  
  
  al get this error 
  
brew test lexido 
==> Testing lexido
==> /home/linuxbrew/.linuxbrew/Cellar/lexido/1.3.1/bin/lexido
2024/04/11 03:42:17 exit status 1
2024/04/11 03:42:18 An error occurred: googleapi: Error 400:
2024/04/11 03:42:18 error details: googleapi: got HTTP response code 400 with body: [{
  "error": {
    "code": 400,
    "message": "API key not valid. Please pass a valid API key.",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "API_KEY_INVALID",
        "domain": "googleapis.com",
        "metadata": {
          "service": "generativelanguage.googleapis.com"
        }
      }
    ]
  }
}
]

Reply:
we got a working one 

`  test do
    # Set a fake API key for testing purposes
    ENV["GOOGLE_AI_KEY"] = "12309863243987"

    # Run the `lexido` command with the "test" argument
    output = shell_output("#{bin}/lexido 2>&1", 1).strip

    # Check if the error message or relevant parts are present in the output
    assert_match "error details: googleapi: got HTTP response code 400 with bod", output
    assert_match "API key not valid. Please pass a valid API key", output
  end`

Reply:
I don't see what is not working. With the wrong API key, it is supposed to throw that error and exit with status 1. It is exactly what we expect

Reply:
> I don't see what is not working. With the wrong API key, it is supposed to throw that error and exit with status 1. It is exactly what we expect

I redirected standard error to standard output (2>&1) and that seemed to work not sure why